apps/web/src/pages/CommunityInterests/CommunityTalentPage/components/CommunityTalentFilterDialog.tsx

```diff
@@ -0,0 +1,293 @@
+  localizedEnumToOptions,
+} from "@gc-digital-talent/forms";
+import { unpackMaybes } from "@gc-digital-talent/helpers";
+import { graphql } from "@gc-digital-talent/graphql";
+
+import FilterDialog, {
+  CommonFilterDialogProps,
+} from "~/components/FilterDialog/FilterDialog";
+import adminMessages from "~/messages/adminMessages";
+import PoolFilterInput from "~/components/PoolFilterInput/PoolFilterInput";
+
+export interface FormValues {
+  communities: string[];
+  workStreams: string[];
+  pools: string[];
+  mobilityInterest: string[];
+  mobilityType: string[];
+  languageAbility: string;
+  employmentDuration: string;
+  workRegions: string[];
+  operationalRequirements: string[];
+  skills: string[];
+}
+
+const context: Partial<OperationContext> = {
+  additionalTypenames: ["Skill", "SkillFamily"], // This lets urql know when to invalidate cache if request returns empty list. https://formidable.com/open-source/urql/docs/basics/document-caching/#document-cache-gotchas
+  requestPolicy: "cache-first", // The list of skills will rarely change, so we override default request policy to avoid unnecessary cache updates.
+};
+
+const CommunityTalentFilterData_Query = graphql(/* GraphQL */ `
+  query CommunityFilterData {
+    communities {
+      id
+      name {
+        localized
+      }
+    }
+    workStreams {
+      id
+      name {
+        localized
+      }
+    }
+    languageAbilities: localizedEnumStrings(enumName: "LanguageAbility") {
+      value
+      label {
+        localized
+      }
+    }
+    workRegions: localizedEnumStrings(enumName: "WorkRegion") {
+      value
+      label {
+        en
+        fr
+      }
+    }
+    operationalRequirements: localizedEnumStrings(
+      enumName: "OperationalRequirement"
+    ) {
+      value
+      label {
+        en
+        fr
+      }
+    }
+    skills {
+      id
+      key
+      name {
+        localized
+      }
+    }
+  }
+`);
+
+type CommunityTalentFilterDialogProps = CommonFilterDialogProps<FormValues>;
+
+const CommunityTalentFilterDialog = ({
+  onSubmit,
+  resetValues,
+  initialValues,
+}: CommunityTalentFilterDialogProps) => {
+  const intl = useIntl();
+
+  const [{ data, fetching }] = useQuery({
+    query: CommunityTalentFilterData_Query,
+    context,
+  });
+
+  const communities = unpackMaybes(data?.communities);
+  const workStreams = unpackMaybes(data?.workStreams);
+  const languageAbilities = unpackMaybes(data?.languageAbilities);
+  const skills = unpackMaybes(data?.skills);
+
+  return (
+    <FilterDialog<FormValues>
+      options={{ defaultValues: initialValues }}
+      {...{ onSubmit, resetValues }}
+    >
+      <div
+        data-h2-display="base(grid)"
+        data-h2-gap="base(x1)"
+        data-h2-grid-template-columns="p-tablet(repeat(2, 1fr))"
+      >
+        <div
+          data-h2-display="base(flex)"
+          data-h2-flex-direction="base(column)"
+          data-h2-gap="base(x1 0)"
+        >
+          <Combobox
```

Comment:
Why is community a filter if the table is meant to "Find GC employees interested in talent mobility opportunities within your community."? Isn't it only meant for a single community?

Reply:
No
The table renders interests for every community the logged in user belongs to.

Reply:
Well, "your community" implies a singular community but 🤷. @NienkeBr, any thoughts?